recipes/libtommath/cmake/conanfile.py

```diff
@@ -0,0 +1,123 @@
+from conan import ConanFile
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class LibTomMathConan(ConanFile):
+    name = "libtommath"
+    description = "LibTomMath is a free open source portable number theoretic multiple-precision integer library written entirely in C."
+    license = "Unlicense"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.libtom.net/"
+    topics = ("math", "mpi", "multi-precision")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return None
+
+    # These are the oldest versions I could find in the docs
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "4.1",
+            "Visual Studio": "8",
+            "clang": "3.3",
+            "apple-clang": "5.0",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        # for plain C projects only
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        # BUILD_SHARED_LIBS and POSITION_INDEPENDENT_CODE are automatically parsed when self.options.shared or self.options.fPIC exist
+        tc = CMakeToolchain(self)
+        # Boolean values are preferred instead of "ON"/"OFF"
+        tc.variables["PACKAGE_CUSTOM_DEFINITION"] = True
```

Comment:
Did you copy the template and not remove anything?

Reply:
I think that's fine, the point of template is to copy from it and adapt for the given library. as long as it passes all the checks, it's  good. it might have some redundancy, and it should be allowed as long as it doesn't break anyone.